File: models/api/Persona.php
<?php

namespace app\models\api;

use Yii;

/**
 * This is the model class for table "personas".
 *
 * @property integer $id
 * @property string $curp
 * @property string $nombres
 * @property string $ape_pat
 * @property string $ape_mat
 * @property string $fecha_nac
 * @property string $sexo
 * @property string $edo_civil
 * @property string $telefono
 * @property string $domicilio
 * @property integer $id_usuario
 *
 * @property Usuarios $idUsuario
 */
class Persona extends \yii\db\ActiveRecord
{
    /**
     * @inheritdoc
     */
    public static function tableName()
    {
        return 'personas';
    }

    /**
     * @inheritdoc
     */
    public function rules()
    {
        return [
            [
                [
                    'curp', 'nombres', 'ape_pat', 'ape_mat', 'fecha_nac',
                    'sexo', 'edo_civil', 'telefono', 'domicilio', 'licencia' ,'id_usuario'
                ],
                'required', 'message' => 'Campo requerido'
            ],
            [
                [
                    'curp', 'nombres', 'ape_pat', 'ape_mat', 'fecha_nac',
                    'sexo', 'edo_civil', 'telefono', 'domicilio', 'licencia' ,'id_usuario'
                ],
                'trim'
            ],
            [
                ['fecha_nac'],
                'date', 'format' => 'yyyy-MM-dd','message' => 'Formato de fecha no valido'
            ],
            [
                ['sexo'], 'in', 'range' => ['M','F'], 'message' => 'Sexo no valido'
            ],
            [
                ['telefono', 'id_usuario'],
                'integer'
            ],
            [
                ['curp'], 'string', 'max' => 18
            ],
            [
                ['nombres'], 'string', 'max' => 40
            ],
            [
                ['ape_pat', 'ape_mat'], 'string', 'max' => 20
            ],
            [
                ['edo_civil'],
                    'in', 'range' => ['casado(a)','soltero(a)'], 'message' => 'Estado civil invalido'
            ],
            [
                ['licencia'],
                    'boolean', 'message' => 'El valor debe ser booleano'
            ],
            [
                ['domicilio'], 'string', 'max' => 150
            ],
            [
                ['curp'], 'unique', 'message' => 'Ya existe un usuario registrado con esta curp'
            ],
            [
                ['id_usuario'],
                    'exist', 'skipOnError' => true,
                        'targetClass' => Usuarios::className(), 'targetAttribute' => ['id_usuario' => 'id']
            ],
            [
                ['id_usuario'],
                    'unique', 'message' => 'Ya existe un registro para este usuario'
            ]
        ];
    }

    /**
     * @inheritdoc
     */
    public function attributeLabels()
    {
        return [
            'id' => 'ID',
            'curp' => 'Curp',
            'nombres' => 'Nombres',
            'ape_pat' => 'Ape Pat',
            'ape_mat' => 'Ape Mat',
            'fecha_nac' => 'Fecha Nac',
            'sexo' => 'Sexo',
            'edo_civil' => 'Edo Civil',
            'telefono' => 'Telefono',
            'domicilio' => 'Domicilio',
            'id_usuario' => 'Id Usuario',
        ];
    }

    /**
     * @return \yii\db\ActiveQuery
     */
    public function getIdUsuario()
    {
        return $this->hasOne(Usuarios::className(), ['id' => 'id_usuario']);
    }
}

File: Android_BolsadeEmpleo/app/src/main/java/com/ittap/bolsadeempleo/models/Persona.java
package com.ittap.bolsadeempleo.models;

import com.ittap.bolsadeempleo.api.SharedPrefManager;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Date;
import java.util.HashMap;


public class Persona {
    public int id = 0;
    public int id_usuario;
    public String curp;
    public String nombres;
    public String ape_pat;
    public String ape_mat;
    public String telefono;
    public String domicilio;
    public String sexo;
    public String fecha_nac;
    public String edo_civil;
    public boolean licencia;

    public Persona( int id_usuario )
    {
        this.id_usuario = id_usuario;
    }

    public Persona( int id, int id_usuario, String curp, String nombres, String ape_pat, String ape_mat, String telefono, String domicilio, String edo_civil, String fecha_nac, boolean licencia)
    {
        this.id = id;
        this.id_usuario = id_usuario;
        this.curp = curp;
        this.nombres = nombres;
        this.ape_pat = ape_pat;
        this.ape_mat = ape_mat;
        this.telefono = telefono;
        this.domicilio = domicilio;
        this.edo_civil = edo_civil;
        this.fecha_nac = fecha_nac;
        this.licencia = licencia;
    }

    public Persona(JSONObject persona) throws JSONException
    {
        id = persona.has("id") ? persona.getInt("id") : 0;
        id_usuario = persona.has("id_usuario") ? persona.getInt("id_usuario") : 0;
        curp = persona.has("curp") ? persona.getString("curp") : "";
        nombres = persona.has("nombres") ? persona.getString("nombres") : "";
        ape_pat = persona.has("ape_pat") ? persona.getString("ape_pat") : "";
        ape_mat = persona.has("ape_mat") ? persona.getString("ape_mat") : "";
        sexo = persona.has("sexo") ? persona.getString("sexo") : "";
        telefono = persona.has("telefono") ? persona.getString("telefono") : "";
        domicilio = persona.has("domicilio") ? persona.getString("domicilio") : "";
        edo_civil = persona.has("edo_civil") ? persona.getString("edo_civil") : "";
        fecha_nac = persona.has("fecha_nac") ? persona.getString("fecha_nac") : "";
        licencia = persona.has("licencia") ? (persona.getInt("licencia") == 1) : false;
    }

    public HashMap<String,String> toHashMap()
    {
        HashMap<String,String> hashMap = new HashMap<>();
        if( id != 0 ) hashMap.put("id", String.valueOf(id));
        hashMap.put("id_usuario", String.valueOf(id_usuario));
        hashMap.put("curp", curp);
        hashMap.put("nombres", nombres);
        hashMap.put("ape_pat", ape_pat);
        hashMap.put("ape_mat", ape_mat);
        hashMap.put("domicilio", domicilio);
        hashMap.put("sexo", sexo);
        hashMap.put("telefono", telefono);
        hashMap.put("edo_civil", edo_civil);
        hashMap.put("fecha_nac", fecha_nac);
        hashMap.put("licencia", licencia ? "1":"0" );

        hashMap.put("tabla","personal");
        return hashMap;
    }

    public HashMap<String,String> toReadHashMap()
    {
        HashMap<String,String> hashMap = new HashMap<>();
        hashMap.put("tabla", "personal");
        hashMap.put("id_usuario", String.valueOf(id_usuario));
        return hashMap;
    }
}
